x-pack/plugins/lens/public/xy_visualization/expression.tsx

```diff
@@ -386,6 +425,19 @@ export function XYChart({
     };
     return style;
   };
+  // Based on the XY toPreviewExpression logic
```

Comment:
Let's move this logic to the `toPreviewExpression` function itself: https://github.com/elastic/kibana/blob/95bfb64ddd725a48d8832ecd978b6895e737d4af/x-pack/plugins/lens/public/xy_visualization/to_expression.ts#L44

Simply overwrite the state setting of show value labels with `hide` in there, then you don't need to detect a preview  expression in here.